.github/workflows/ci.yml

```
@@ -13,10 +13,64 @@ on:
       - release/**
 
 jobs:
+# add get-diff job 
+  get-diff:
+    runs-on: ubuntu-latest
+    # Map a step output to a job output
+    outputs:
+      contract: ${{ steps.get-diff-contract.outputs.diff }}
+      certora-contract: ${{ steps.get-diff-certora-contract.outputs.diff }}
+      scripts: ${{ steps.get-diff-scripts.outputs.diff }}
+      agents: ${{ steps.get-diff-agents.outputs.diff }}
+      env: ${{ steps.get-diff-env.outputs.diff }}
+    
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Get diff for contract
+        uses: technote-space/get-diff-action@v4.0.1
+        id: get-diff-contract # via outputs
+        with:
+          PATTERNS: |
+            +(contracts)/*.sol
+
+      - name: Get diff for certora contract
+        uses: technote-space/get-diff-action@v4.0.1
+        id: get-diff-certora-contract # via outputs
+        with:
+          PATTERNS: |
+            +(certora)/**/*.sol
+      
+      - name: Get diff for scripts/test
+        uses: technote-space/get-diff-action@v4.0.1
+        id: get-diff-scripts # via outputs
+        with:
+          PATTERNS: |
+            +(scripts/test)/*.+(js|ts)
+            +(scripts/test)/**/*.+(js|ts)
```

```suggestion
            +(scripts|test)/*.+(js|ts)
            +(scripts|test)/**/*.+(js|ts)
```
small but important details

---

Done